ADR/adr-0008-environment-provisioning.md

```diff
@@ -0,0 +1,428 @@
+            cpu: 8m
+    arch: x86–64
+    kubernetesCredentials:
+        defaultNamespace: team-a--prod-dtc
+        apiURL: …
+        clusterCredentialsSecret: team-a--prod-dtc--secret
+    claimRef:
+        name: prod-dtc
+```
+
+#### DeploymentTargetClaim (DTC)
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: DeploymentTargetClaim
+metadata:
+    name: prod-dtc
+spec:
+    deploymentTargetClassName: isolation-level-namespace
+    resources:
+        requests:
+            memory: 16Gi
+            cpu: 8m
+        limits:
+            memory: 16Gi
+            cpu: 8m
+    arch: x86–64
+status:
+    phase: Bound
+```
+
+#### DeploymentTargetClass (DTCLS)
+
+Hypershift cluster with 3 nodes example:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: DeploymentTargetClass
+metadata:
+    name: isolation-level-cluster-small
+spec:
+    provisioner: appstudio.redhat.com/hypershift
+    parameters:
+        numOfNodes: 3
+    reclaimPolicy: Delete
+    allowTargetResourceExpansion: false
+```
+
+### Phase 3
+
+The result of this phase is to automatically provisioning of Hypershift cluster using Stonesoup
+credentials. We call it “provided compute” (compute that we provide, not the user) and it’s included
+as part of the offering. This compute can be used for both long lived clusters and for ephemeral
+clusters used by the integration service.
+
+For long lived clusters, the maintenance model for them is yet to be determined.
+
+#### DeploymentTargetClass (DTCLS)
+
+Hypershift cluster with 3 nodes example:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: DeploymentTargetClass
+metadata:
+    name: isolation-level-cluster-small
+spec:
+    provisioner: appstudio.redhat.com/hypershift
+    parameters:
+        numOfNodes: 3
+    reclaimPolicy: Delete
+    allowTargetResourceExpansion: false
+```
+
+Hypershift cluster with 6 nodes example:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: DeploymentTargetClass
+metadata:
+    name: isolation-level-cluster-large
+spec:
+    provisioner: appstudio.redhat.com/hypershift
+    parameters:
+        numOfNodes: 6
+    reclaimPolicy: Delete
+    allowTargetResourceExpansion: false
+```
+
+## References
+
+- [Discussion recording, 2022-12-07](https://drive.google.com/file/d/1UWIefLzFb2rYnz8mvvWkNKJyuWy-_nhm/view)
+- [Original drafting in googledoc](https://docs.google.com/document/d/1vFD5lDbxek_Q2RZyKdFuZ8z0xYubOA1i8cu2K7KNxaQ/edit#heading=h.k2fng7c0dm7k)
+- [Original miro](https://miro.com/app/board/uXjVP77ztI4=)
+- [Issue tracking creation of this ADR (STONE-174)](https://issues.redhat.com/browse/STONE-174)
+
+[Environment]: ../ref/application-environment-api.md#environment
+[Environments]: ../ref/application-environment-api.md#environment
+[DT]: ../ref/application-environment-api.md#deploymenttarget
+[DTs]: ../ref/application-environment-api.md#deploymenttarget
+[DeploymentTarget]: ../ref/application-environment-api.md#deploymenttarget
```

Comment:
This url anchor won't be valid until something actually gets implemented in the application-api repo.